### ISSUE TITLE
Internal error - failed to start - couldn't start client ActionScrip & MXML Language Server

### ISSUE THREAD

**Issue by mjesteve** [BowlerHatLLC/vscode-as3mxml#524]
Yesterday, I was working normally with a JSRoyale project and suddenly the extension stopped working. It showed an Internal error.
I had updated the vscode to version 1.53 and I thought that could be the cause but it seems that it is not (I have recovered an old pc I have updated to the same version and everything works fine, also other colleagues from the Apache Royale forum have told me that to they also work)
The first errors I got:
- When I opened the project:
![image](https://user-images.githubusercontent.com/55754195/107082620-bcabf800-67f4-11eb-8c29-599b9861f9e1.png)

- In **DevTools** you could see:
![ErrorVSCode](https://user-images.githubusercontent.com/55754195/107082765-f7159500-67f4-11eb-843f-3e33dbb6166f.png)

I uninstalled all the extensions and reinstalled them.
I updated the java version to 1.8.0.281 (before I had 231)
Everything was still not working.
I've uninstalled vscode and reinstalled it, and now the error still occurs but with slightly different errors:
![image](https://user-images.githubusercontent.com/55754195/107089088-51672380-67fe-11eb-880a-a73cf4ab7ad1.png)

And in **OUTPUT**:
**Log(Extension Host)**

[2021-02-05 20:56:56.098] [exthost] [info] extension host started
[2021-02-05 20:56:56.191] [exthost] [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication {"startup":false,"extensionId":{"value":"vscode.microsoft-authentication","_lower":"vscode.microsoft-authentication"},"activationEvent":"onAuthenticationRequest:microsoft"}
[2021-02-05 20:56:56.192] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/microsoft-authentication/dist/extension.js
[2021-02-05 20:56:56.200] [exthost] [info] ExtensionService#_doActivateExtension vscode.css-language-features {"startup":false,"extensionId":{"value":"vscode.css-language-features","_lower":"vscode.css-language-features"},"activationEvent":"onLanguage:css"}
[2021-02-05 20:56:56.200] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/css-language-features/client/dist/node/cssClientMain
[2021-02-05 20:56:56.209] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet {"startup":false,"extensionId":{"value":"vscode.emmet","_lower":"vscode.emmet"},"activationEvent":"onLanguage:css"}
[2021-02-05 20:56:56.209] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/emmet/dist/node/emmetNodeMain
[2021-02-05 20:56:56.215] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch {"startup":true,"extensionId":{"value":"vscode.debug-auto-launch","_lower":"vscode.debug-auto-launch"},"activationEvent":"*"}
[2021-02-05 20:56:56.215] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/debug-auto-launch/dist/extension
[2021-02-05 20:56:56.218] [exthost] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"extensionId":{"value":"vscode.github","_lower":"vscode.github"},"activationEvent":"*"}
[2021-02-05 20:56:56.218] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/git/dist/main
[2021-02-05 20:56:56.328] [exthost] [info] ExtensionService#_doActivateExtension vscode.github {"startup":true,"extensionId":{"value":"vscode.github","_lower":"vscode.github"},"activationEvent":"*"}
[2021-02-05 20:56:56.328] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/github/dist/extension.js
[2021-02-05 20:56:56.333] [exthost] [info] ExtensionService#_doActivateExtension ms-vscode.js-debug {"startup":false,"extensionId":{"value":"ms-vscode.js-debug","_lower":"ms-vscode.js-debug"},"activationEvent":"onCommand:extension.js-debug.clearAutoAttachVariables"}
[2021-02-05 20:56:56.333] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/ms-vscode.js-debug/src/extension.js
[2021-02-05 20:56:56.394] [exthost] [info] ExtensionService#_doActivateExtension bowlerhatllc.vscode-swf-debug {"startup":true,"extensionId":{"value":"bowlerhatllc.vscode-nextgenas","_lower":"bowlerhatllc.vscode-nextgenas"},"activationEvent":"workspaceContains:**/asconfig.json,**/asconfig.*.json,*.as3proj"}
[2021-02-05 20:56:56.394] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/.vscode/extensions/bowlerhatllc.vscode-swf-debug-1.2.2/extension
[2021-02-05 20:56:56.400] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication {"startup":false,"extensionId":{"value":"vscode.github-authentication","_lower":"vscode.github-authentication"},"activationEvent":"onAuthenticationRequest:github"}
[2021-02-05 20:56:56.400] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/github-authentication/dist/extension.js
[2021-02-05 20:56:56.410] [exthost] [info] ExtensionService#_doActivateExtension bowlerhatllc.vscode-nextgenas {"startup":true,"extensionId":{"value":"bowlerhatllc.vscode-nextgenas","_lower":"bowlerhatllc.vscode-nextgenas"},"activationEvent":"workspaceContains:**/asconfig.json,**/asconfig.*.json,*.as3proj"}
[2021-02-05 20:56:56.410] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/.vscode/extensions/bowlerhatllc.vscode-nextgenas-1.5.0/extension
[2021-02-05 20:56:56.666] [exthost] [info] eager extensions activated
[2021-02-05 20:56:56.694] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":false,"extensionId":{"value":"vscode.merge-conflict","_lower":"vscode.merge-conflict"},"activationEvent":"onStartupFinished"}
[2021-02-05 20:56:56.694] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/merge-conflict/dist/mergeConflictMain
[2021-02-05 20:56:56.698] [exthost] [info] ExtensionService#_doActivateExtension ms-vscode-remote.remote-wsl-recommender {"startup":false,"extensionId":{"value":"ms-vscode-remote.remote-wsl-recommender","_lower":"ms-vscode-remote.remote-wsl-recommender"},"activationEvent":"onStartupFinished"}
[2021-02-05 20:56:56.698] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/ms-vscode-remote.remote-wsl-recommender/dist/extension.js
[2021-02-05 20:56:56.704] [exthost] [info] ExtensionService#_doActivateExtension vscode.testing-editor-contributions {"startup":false,"extensionId":{"value":"vscode.testing-editor-contributions","_lower":"vscode.testing-editor-contributions"},"activationEvent":"onStartupFinished"}
[2021-02-05 20:56:56.704] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/hiedra/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/testing-editor-contributions/dist/extension.js

Any ideas?
Any help will be welcome.

Thx.
Hiedra

**Comment by joshtynjala** [comment]
This is likely caused by a Java exception during the startup of the language server. Unfortunately, that's tricky to debug because you don't get nice stack traces from exceptions in the language server until after a successful startup.

I saw on the Royale mailing list that you use the `as3mxml.sdk.editor` setting. Do you get the same error if you clear that setting, and use the default?

**Comment by mjesteve** [comment]
Hi Josh, Yes, the same mistake.

I activated the java console but did not see anything (I am not an expert in java and its debugging, the only thing I have done has been a scan applet and I did see the errors in it)

This is my setup:
![image](https://user-images.githubusercontent.com/55754195/107097364-16b8b780-680d-11eb-941c-9b704e7c1bde.png)

![image](https://user-images.githubusercontent.com/55754195/107097405-289a5a80-680d-11eb-848d-85db301f0545.png)

If you tell me that it may be a java problem, I will focus there. Before I updated the version, now I'm going to uninstall Java and I'll tell you if I have progress, hahaha, and if I have no progress too :)

Thx josh,
Hiedra

**Comment by joshtynjala** [comment]
It could be that a recent change to your project is causing this issue. I would try reverting to an earlier commit of your project.

You might also try reverting to an earlier commit of royale-asjs to see if a recent change there is the cause. royale-compiler hasn't changed in at least 16 days, so unless you hadn't updated it recently, I don't think that any recent changes there would have caused this issue.

This may be difficult to debug, so I think that the best way forward is to try to determine what changed in your environment recently. If you can get yourself into a working state again, it'll be easier to narrow it down to a specific commit in either your project or royale-asjs (or possibly something else).

**Comment by mjesteve** [comment]
Nothing Josh, everything the same. Now I only have version 1.8.0.281 installed but everything remains the same.
When I uninstalled, I rebooted the PC and opened the vscode to verify that the extension requested the vm and effectively:
![image](https://user-images.githubusercontent.com/55754195/107098411-6c8e5f00-680f-11eb-92d8-fe3090edadc4.png)

But, after installation, it gives the same errors again and the OUTPUT is also the same.
![image](https://user-images.githubusercontent.com/55754195/107098489-9c3d6700-680f-11eb-9abf-1e17ea71f0f8.png)

(While I was assembling this message, the last one of yours has reached me ... I answer you)
The error occurs to me in any project. I have tried opening all the projects I have ranging from a simple Hello example to royale sdk.
What I have not tried is to compile the sdk in a previous commit, what I have tried is to configure the 0.9.7 distribution of the sdk with several projects and the errors are the same.
I'm going to compile the sdk and tell you.
Hiedra

**Comment by mjesteve** [comment]
Eureka !!! You're a crack Josh "a breakthrough" ... I can't quite reason how you've put all the pieces together hahaha (java - compiler - sdk ...)
After several compilations of the sdk, I downloaded from jenkis the last successful distribution of January 1, and I opened my project "without problems". (I don't know why version 0.9.7 also gave me an error ... maybe it was a distribution compiled by me instead of a distribution)

To verify that the sdk is the source of the error I have carried out several tests (with my project and with the TDJ):
- wk settings configured with latest jenkins build (# 2436) ==> OK
    "settings": {
        "as3mxml.sdk.framework": "D: \\ Apache \\ apache-royale-0.9.8-bin-js \\ royale-asjs",
        "as3mxml.sdk.editor": "D: \\ Apache \\ apache-royale-0.9.8-bin-js \\ royale-asjs"
    }
- wk settings configured with my last build yesterday maven + ant ==> ERROR
- wk settings configured with my last build today maven ==> (I could not test it because it gives me an error that I will investigate now: "No compiler is provided in this environment. Perhaps you are running on a JRE rather than a JDK?")

The error is clear that it occurs with my SDK builds, so far I have never had problems and my co-worker works everything without problems.

I will see investigate ...
Thank you very much Josh, when I can try my compilation today I will tell you more things.
If you are seeing something that I am not seeing, please tell me.

Hiedra

**Comment by mjesteve** [comment]
I couldn't get the vscode to work by compiling my fork so I deleted everything and cloned the repository again.
After compiling with maven (with the -P withdistribution option) and then with ant I have been successful and have not experienced the error again.
I would have liked to find out the cause of the error as well as why from one moment to another what had been working for me for several months stopped working, maybe I did something and I don't remember :(

Josh thank you very much for your patience and your help.
See you on the Royale forum.

Thx.
Hiedra.
